library/lists.pl

```diff
@@ -552,6 +553,21 @@
     Sum1 is Sum0 + X,
     sum_list(Xs, Sum1, Sum).
 
+%!  prod_list(+List, -Sum) is det.
+%
+%   Prod is the result of multiplying all numbers in List.
+
+prod_list(Xs, Prod) :-
```

Comment:
I suggest you add some test cases.

For example:
```
?- debug, prod_list([1,2,0,3], Z).
ERROR: Arguments are not sufficiently instantiated
ERROR: In:
ERROR:   [16] _8848 is _8854*3
ERROR:   [15] prod_list([3],_8882,0) at /tmp/test_prod.pl:5
ERROR:   [14] prod_list([0,3],2,0) at /tmp/test_prod.pl:10
ERROR:   [13] prod_list([2,0|...],1,0) at /tmp/test_prod.pl:10
ERROR:   [12] prod_list([1,2|...],1,0) at /tmp/test_prod.pl:10
ERROR:   [11] prod_list([1,2|...],0) at /tmp/test_prod.pl:2
ERROR:   [10] '<meta-call>'(user:user: ...) <foreign>
ERROR:    [9] toplevel_call(user:user: ...) at /usr/lib/swi-prolog/boot/toplevel.pl:1113
```

Reply:
Thanks, this was a silly mistake